app/user.go

```diff
@@ -1207,6 +1213,20 @@ func (a *App) GetPasswordRecoveryToken(token string) (*model.Token, *model.AppEr
 	return rtoken, nil
 }
 
+func (a *App) DeletePasswordRecoveryTokenForUser(userId string) *model.AppError {
```

Comment:
Wondering if instead of deleting the tokens, we just associate the tokens with the email address? That way if your email changes they are automatically invalid without having to run a bunch of deletes.

Reply:
Actually that sounds like a way better idea. Through the normal token clean up they will get automatically deleted after X hours anyway. I'll rework the PR to follow that path.

Reply:
Implemented & pushed